pkg/storage/loaders/model_plan_tracking_date_loader.go

```diff
@@ -0,0 +1,50 @@
+package loaders
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/graph-gophers/dataloader"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/mint-app/pkg/appcontext"
+)
+
+// GetModelPlanTrackingDateByModelPlanID uses a DataLoader to aggregate a SQL call and return the Model Plan Tracking Date in one query
+func (loaders *DataLoaders) GetModelPlanTrackingDateByModelPlanID(ctx context.Context, keys dataloader.Keys) []*dataloader.Result {
+	dr := loaders.DataReader
+
+	logger := appcontext.ZLogger(ctx)
+	arrayCK, err := ConvertToKeyArgsArray(keys)
+	if err != nil {
+		logger.Error("issue converting keys for data loader in Model Plan Tracking Date", zap.Error(*err))
+		return nil
+	}
+	marshaledParams, err := arrayCK.ToJSONArray()
+	if err != nil {
+		logger.Error("issue converting keys to JSON for data loader in Model Plan Tracking Date", zap.Error(*err))
+		return nil
+	}
+
+	trackingDates, _ := dr.Store.ModelPlanTrackingDateGetByModelPlanIDLOADER(logger, marshaledParams)
```

Comment:
Could you handle this error here please?

Reply:
Yes. It's worth considering that we currently do not capture or handle the error from model_plan_loader.go:29:

	`plans, _ := dr.Store.ModelPlanGetByModelPlanIDLOADER(logger, marshaledParams)`
	
We may benefit from eventual static analysis to catch other such unhandled error situations.

I previously added error handling, but removed it due a capturing error. It seems for some reason we pass error pointers in prior utility methods, which is not idiomatic to Go. Part of tech investment should likely be to restructure these methods to return a simple error struct. I will work around this in the moment by creating an `err2` value.

Reply:
I think you are right, I forget the reason those helper methods returned a pointer, but I do agree it probably should just be the value instead of a pointer.